src/lib/Liblog/pbs_log.c

```diff
@@ -386,21 +386,21 @@ log_add_debug_info()
 	/* To add leaf node name, if set */
 	strcpy(temp, "pbs_leaf_name=");
 	if(pbs_conf.pbs_leaf_name){
-		strncat(temp, pbs_conf.pbs_leaf_name, LOG_BUF_SIZE);
+		strncat(temp, pbs_conf.pbs_leaf_name, strlen(temp)+strlen(pbs_conf.pbs_leaf_name));
```

Comment:
Why are we bothering to calculate string length here? We already know how large the buffer is. On the next line we call strncat() again using the LOG_BUF_SIZE macro.

We should also ensure the string is terminated... temp[LOG_BUF_SIZE - 1] = '\0';

You might also choose to use snprintf() because it will terminate the string for you.

Reply:
I vote for the use of snprintf().  Using strncat() may or may not take longer than snprintf().  While snprintf() has to do more parsing of the format string, strncat() writes N bytes.  It will write the string, and the rest of the buffer is zeroed.